### DISCUSSION TITLE
Support to add Your Own DB

### DISCUSSION THREAD

**Discussion by vinaayakha** [mnfst/llm-gateway#100] [Feature request]
Hi Case Community,

Love the implementation of BaaS in Case.

I'm a Dev and I like to have control of how to name the routes and use existing tables to create entities in Case and allow my non-dev internal team with correct rights to update the models from the UI and filter the data that is required in the API.

Is it something that is in the pipeline?

And If it is can I also be involved in it?

Thanks

**Comment by brunobuddy** [comment]
Hello @vinaayakha,

Thank you for your message ! I am going to answer each topic individually:

- **Name routes**
Routes are based upon entity slugs, that you can edit in the entity files passing [entity params](https://docs.case.app/entities?id=entity-params) to the `@Entity()` decorator. Ex:
```js
@Entity({
   slug: "my-custom-slug"
})
```
By default, the slug in the "pluralized" name of the entity class name (REST convention)

- **Import DB**
Not implemented. It is kind of harsh as every DB has its own specs. CASE uses TypeORM and someone made [a package](https://github.com/Kononnable/typeorm-model-generator) for this (not tested). The result should be close to what we have in CASE

- **Visual structure editing**
This feature is not planned to be implemented in the future. CASE's approach is "pro-code", which gives advantages to devs: code sharing, easier versioning and less lock-in. Most of the BaaS in the market have a visual approach, which can be cumbersome to some developers.

- **How can you contribute ?** 
Thanks for your interest ! As CASE is still in BETA, we are looking for people's feedback on existing features and how we can improve DX on the current product. A feature or a doc that did not work as expected or that failed, looking for inconsistencies in the experience, things that are not very clear, etc... Feel free to open issues or do a PR, we would be very pleased 😄
